crates/bevy_ecs/src/component.rs

```diff
@@ -138,6 +138,24 @@ impl ComponentInfo {
     }
 }
 
+/// A [`ComponentId`] is an semi-opaque value which uniquely identifies the type of
+/// a [`Component`] within a [`World`](crate::world::World).
+///
+/// Each time a new [`Component`] type is registered within a [`World`](crate::world::World)
+/// using [`World::init_component`](crate::world::World::init_component) or
+/// [`World::init_component_with_descriptor`](crate::world::World::init_component_with_descriptor),
+/// a corresponding [`ComponentId`] is created to track it.
```

Comment:
```suggestion
/// a corresponding `ComponentId` is created to track it.
```

Link to self is not useful, and drags away the attention towards the other links.